### DISCUSSION TITLE
Sudden freeze of teslamate

### DISCUSSION THREAD

**Discussion by djpalmis** [teslamate-org/teslamate#3958] [Q&A]
While I was charging during the night it stopped refreshing data.

I'm not so skilled about teslamate logs but this is what i get:
```
_teslamate_1  | 2024-06-14 09:08:15.546 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/NUMBERHIDED/vehicle_data -> 429 (77.801 ms)
_teslamate_1  | 2024-06-14 09:08:15.546 [warning] TeslaApi.Error / "Retry in 60705 seconds\n"
_teslamate_1  | 2024-06-14 09:08:15.546 car_id=1 [error] Error / :unknown
_teslamate_1  | 2024-06-14 09:08:30.634 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/NUMBERHIDED/vehicle_data -> 429 (86.559 ms)
_teslamate_1  | 2024-06-14 09:08:30.634 [warning] TeslaApi.Error / "Retry in 60690 seconds\n"
_teslamate_1  | 2024-06-14 09:08:30.634 car_id=1 [error] Error / :unknown
```
what shall I do? Shall I restart the virtual machine?
Car now is waiting to sleep

Edit: A bug ticket is opened https://github.com/teslamate-org/teslamate/issues/3957

**Comment by djpalmis** [comment]
I restarted the virtual machine, now it seems that it is update (it says car is offline)

**Comment by djpalmis** [comment]
Data is lost since the last drive.
It still does not connect to my car!

**Comment by djpalmis** [comment]
It is ok now after many tries. I can't understand what was wrong and what fixed it. 
Maybe car deep sleep is the most significant change.

**Comment by djpalmis** [comment]
Broke again. When I am in the car the new "API use icon" is not showed. So I understand that teslamate is not getting data from car.
I don't know what's happening. 
Maybe it is an issue car side, maybe server side.

**Comment by browellm** [comment]
I think there's something going on with the API - Octopus was giving out some charging slots this morning thinking my car's SoC was lower than it actually is.

**Comment by ToniA** [comment]
Starting today, this is 2016 Model S with MCU1
```
2024-06-15 13:04:52.764 [warning] WebSocket disconnected. Reconnecting …
2024-06-15 13:09:27.358 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/xxxxx/vehicle_data -> 429 (96.720 ms)
2024-06-15 13:09:27.358 [warning] TeslaApi.Error / "Retry in 49833 seconds\n"
2024-06-15 13:09:27.359 car_id=1 [error] Error / :unknown
```

**Comment by ToniA** [comment]
Seems that we are now rate-limited...

```
2024-06-15 14:39:23.876 [warning] TeslaApi.Error / %TeslaApi.Error{
  reason: :unknown,
  message: "\"Retry in 44437 seconds\\n\"",
  env: %Tesla.Env{
    method: :get,
    url: "https://owner-api.teslamotors.com/api/1/vehicles/xxxxx/vehicle_data",
    query: [
      endpoints: "charge_state;climate_state;closures_state;drive_state;gui_settings;location_data;vehicle_config;vehicle_state;vehicle_data_combo"
    ],
    headers: [
      {"content-type", "text/plain"},
      {"ratelimit-limit", "600"},
      {"ratelimit-remaining", "0"},
      {"ratelimit-reset", "44437"},
      {"retry-after", "44437"},
      {"x-request-id", "f33d1201-dd2f-4b15-b08e-798609bddc5e"},
      {"vary", "Accept-Encoding"},
      {"x-envoy-upstream-service-time", "18"},
      {"x-frame-options", "DENY"},
      {"x-content-type-options", "nosniff"},
      {"strict-transport-security", "max-age=31536000; includeSubDomains"},
      {"cache-control", "no-cache, no-store, private, s-max-age=0"},
      {"date", "Sat, 15 Jun 2024 11:39:23 GMT"},
      {"server", "envoy"},
      {"transfer-encoding", "chunked"}
    ],
    body: "Retry in 44437 seconds\n",
    status: 429,
    opts: [
      access_token: "xxxxx"
    ],
    __module__: TeslaApi,
    __client__: %Tesla.Client{fun: nil, pre: [], post: [], adapter: nil}
  }
}
```

**Reply by binou06** [reply]
Same here for my XPlaid 2023 (just updated in 2024.20.1.83) :
`/var/log/syslog:Jun 15 14:11:44 tesla teslamate[13207]: 2024-06-15 14:11:44.893 [warning] TeslaApi.Error / "Retry in 42496 seconds\n"
`

But my wife Y Perf (2024.14.8) is still ok (still broadcasting infos to my teslamate)

**Comment by djpalmis** [comment]
Teslamate still broken. No way to get data again. What about teslafi? Does it work?

**Comment by binou06** [comment]
After the countdown (Retry in xxx seconds), it works again. But strange behaviour, before, the speed / location was updated every couple of seconds but here, it seems to be updated every, let say, 15seconds. Is the websocket does not give as much info as before or not at all (and teslamate falls back in old mode - like before websocket when it pulls info) ?

**Comment by djpalmis** [comment]
It works ok since 2 days without any change by my side.
So.. Was it just a temporary issue? Was it a warning by Tesla?

**Reply by binou06** [reply]
Perhaps a try ? I think that Tesla shoud think about us (let be foul, give a special API key for Teslamate users). Lots of Tesla owners are geeks. Access to API is for me a must have because I use it to self open my gate, switch on lights, and to display infos on my Nest Hubs to organize my life (if my wife comes back home, it displays infos about car and if I forgot to empty the dish washer, I ran to do it 🤣). Jokes appart, it is a game change to organize my life. And these examples are juste an little part of what I do (for instance, as we park in reverse, if one of our car is in reverse in haley, it switch off house alarm, unlock the door, ... If one of our car is plugged in and the solar is over-generating, I force the charge and adapt the power to follow the energy produced and not consumed.

Me and my wife have a Tesla (a Y and a X) and it is the 5th Tesla we buy. When I last change my car, I start to think about other brands. One of the things that keep me in Tesla's word is ability of using API and Teslamate. I think that if API no longer work that way (I am ok to pay but not for 1 request every 5 minutes !), I'll have a look at Porsche for instance because Tesla still suffer in quality. My X is 10 months old and the Yoke has already scratches. I saw on the internet that lots of people have the same and Tesla says (after 10months) that it is "normal". For a 140k€ car it is a shame. 

I know I'm away from the topic but a lot of people use Teslamate but there are not so much comments on this github. I hope some Tesla emploies are reading theses threads and will carry our voice.

**Comment by GPSBabelDeveloper** [comment]
@binou06, Tesla is not going to give you a special API or key.  That's not happening. They've long had a love/hate relationship with 3rd party apps in general. (Hint: they create a lot of battery-related support traffic for them, ranging from the app legitimately eating the battery to people pestering support centers that their battery dropped 1% over the last year.)

On the bright side, the headline feature of
  https://github.com/teslamate-org/teslamate/releases/tag/v1.29.2 
  is "improve TeslaMate behavior on Tesla API rate limits". Look like that'll help you and @ToniA and a few others here.
  
On the down side, another open [issue](https://github.com/teslamate-org/teslamate/discussions/3975) says that 1.29 is slow. So personally, I'm going to wait it out a bit. (I totally get that if everyone does that, there's no good data on going forward.)


I'm also motivated to wait because my TeslaMate Pi ate (another) "endurance" card, and had already re-installed TeslaMate, but on my Synology where hopefully the write traffic won't eat the car, and just added another Tesla to my stable and now (stupidly) have a mix of some data that I can recover and some new data that I don't want to lose, so I'll play database surgeon, notably altering car IDs since I now have three DIFFERENT IDs in various tables. :-| Nothing a little awk/grep/perl won't fix.

But beyond changing OSes, running on different hardware, having slightly broken backups, deciding if I want to try to pg_dump the volumes or just use Jan's backup (the Pi died in March), adding a Tesla, and changing Teslamate versions, I don't have any variables and everything else is the same. Honest! :-)

**Comment by ToniA** [comment]
I first updated to the 'edge' build (from 'latest'). Co-incidence or not, but this started to work fine after the retry-after had elapsed. I just switched back to 'latest', pulled, and got 1.29.2 as expected. So far so good :)
